core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/UndirectedInfraBuilder.java

```diff
@@ -168,13 +171,62 @@ private TrackSectionImpl makeTrackSection(RJSTrackSection track) {
                 track.id,
                 ImmutableSet.copyOf(operationalPointsPerTrack.get(track.id)),
                 track.geo,
-                track.sch
+                track.sch,
+                buildLoadingGaugeLimits(track.loadingGaugeLimits)
         );
         builder.addEdge(begin, end, edge);
         edge.gradients = makeGradients(track);
         return edge;
     }
 
+    /** Builds the ranges of blocked loading gauge types on the track */
+    private ImmutableRangeMap<Double, LoadingGaugeConstraint> buildLoadingGaugeLimits(
+            List<RJSLoadingGaugeLimit> limits
+    ) {
+        // This method has a bad complexity compared to more advanced solutions,
+        // but we don't expect more than a few ranges per section
+        if (limits == null)
+            return ImmutableRangeMap.of();
+        var builder = new ImmutableRangeMap.Builder<Double, LoadingGaugeConstraint>();
+        var transitions = new TreeSet<Double>();
+        for (var range : limits) {
+            transitions.add(range.begin);
+            transitions.add(range.end);
+        }
+        var transitionsList = new ArrayList<>(transitions);
+        for (int i = 1; i < transitionsList.size(); i++) {
+            var begin = transitionsList.get(i - 1);
+            var end = transitionsList.get(i);
+            var allowedTypes = new HashSet<RJSLoadingGaugeType>();
+            for (var range : limits)
+                if (range.begin <= begin && range.end >= end)
+                    allowedTypes.addAll(getCompatibleGaugeTypes(range.type));
+            var blockedTypes = Sets.difference(
+                    Sets.newHashSet(RJSLoadingGaugeType.values()),
+                    allowedTypes
+            );
+            builder.put(Range.open(begin, end), new LoadingGaugeConstraintImpl(ImmutableSet.copyOf(blockedTypes)));
+        }
```

Comment:
please add a comment like "TODO: use an interval tree"
the logic is pretty weird, it all makes sense but in the end emulates adding ranges to an interval tree and iterating over these